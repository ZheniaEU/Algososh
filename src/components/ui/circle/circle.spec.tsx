import { render } from "@testing-library/react"
import { ElementStates } from "types/element-states"
import { Button } from "../button/button"
import { Circle } from "./circle"

describe("Circle snapshot tests", () => {

   it("Circle without props", () => {
      const { asFragment } = render(<Circle />)
      expect(asFragment()).toMatchSnapshot()
   })

   it("Circle with letter as number", () => {
      const { asFragment } = render(<Circle letter={5} />)
      expect(asFragment()).toMatchSnapshot()
   })

   it("Circle with letter as string", () => {
      const { asFragment } = render(<Circle letter={"5"} />)
      expect(asFragment()).toMatchSnapshot()
   })

   it("Circle with tail", () => {
      const { asFragment } = render(<Circle letter={"tail"} />)
      expect(asFragment()).toMatchSnapshot()
   })

   it("Circle with head", () => {
      const { asFragment } = render(<Circle head={"head"} />)
      expect(asFragment()).toMatchSnapshot()
   })

   //но зачем? этого нет в проектной, это бессмысленно и не собержит функционала
   it("Circle with head react-element", () => {
      const { asFragment } = render(<Circle head={<Button />} />)
      expect(asFragment()).toMatchSnapshot()
   })

   //но зачем? этого нет в проектной, это бессмысленно и не собержит функционала
   it("Circle with tail react-element", () => {
      const { asFragment } = render(<Circle tail={<Button />} />)
      expect(asFragment()).toMatchSnapshot()
   })

   it("Circle with index", () => {
      const { asFragment } = render(<Circle index={5} />)
      expect(asFragment()).toMatchSnapshot()
   })

   it("Circle with isSmall === true", () => {
      const { asFragment } = render(<Circle isSmall={true} />)
      expect(asFragment()).toMatchSnapshot()
   })

   it("Circle with state default", () => {
      const { asFragment } = render(<Circle state={ElementStates.Default} />)
      expect(asFragment()).toMatchSnapshot()
   })

   it("Circle with state changing", () => {
      const { asFragment } = render(<Circle state={ElementStates.Changing} />)
      expect(asFragment()).toMatchSnapshot()
   })

   it("Circle with state modified", () => {
      const { asFragment } = render(<Circle state={ElementStates.Modified} />)
      expect(asFragment()).toMatchSnapshot()
   })

})
