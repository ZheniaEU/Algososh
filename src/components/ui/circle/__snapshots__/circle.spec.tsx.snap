// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle snapshot tests Circle with head 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head string"
      data-testid="head"
    >
      head
    </div>
    <div
      class="circle   default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
        data-testid="letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="tail"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle snapshot tests Circle with head react-element 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="head"
    >
      <button
        class="text_type_button text_color_primary button undefined false "
        type="button"
      >
        <p
          class="text undefined"
        />
      </button>
    </div>
    <div
      class="circle   default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
        data-testid="letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="tail"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle snapshot tests Circle with index 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="head"
    />
    <div
      class="circle   default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
        data-testid="letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    >
      5
    </p>
    <div
      class="text text_type_input text_color_input mt-4 absolute tail60 element"
      data-testid="tail"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle snapshot tests Circle with isSmall === true 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="head"
    />
    <div
      class="circle  small default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
        data-testid="letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="tail"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle snapshot tests Circle with letter as number 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="head"
    />
    <div
      class="circle   default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
        data-testid="letter"
      >
        5
      </p>
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="tail"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle snapshot tests Circle with letter as string 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="head"
    />
    <div
      class="circle   default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
        data-testid="letter"
      >
        5
      </p>
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="tail"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle snapshot tests Circle with state changing 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="head"
    />
    <div
      class="circle   changing"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
        data-testid="letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="tail"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle snapshot tests Circle with state default 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="head"
    />
    <div
      class="circle   default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
        data-testid="letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="tail"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle snapshot tests Circle with state modified 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="head"
    />
    <div
      class="circle   modified"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
        data-testid="letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="tail"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle snapshot tests Circle with tail 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="head"
    />
    <div
      class="circle   default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
        data-testid="letter"
      >
        tail
      </p>
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="tail"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle snapshot tests Circle with tail react-element 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="head"
    />
    <div
      class="circle   default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
        data-testid="letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="tail"
    >
      <button
        class="text_type_button text_color_primary button undefined false "
        type="button"
      >
        <p
          class="text undefined"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Circle snapshot tests Circle without props 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="head"
    />
    <div
      class="circle   default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
        data-testid="letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="tail"
    />
  </div>
</DocumentFragment>
`;
